src/components/Api.js

```diff
@@ -0,0 +1,115 @@
+import { routeData as routes } from "../utils/constants";
+const debug = false;
+// if(debug) console.log(``);
+
+export default class Api {
+  constructor({ baseUrl, headers }) {
+    this._baseUrl = baseUrl;
+    this._authorization = headers.authorization;
+    this._contentType = headers["Content-Type"];
+  }
+
+  _buildRequest(routeData, body = undefined) {
+    const url = this._baseUrl + routeData.route;
+    const info = {
+      method: routeData.method,
+      headers: {
+        authorization: this._authorization,
+        "Content-Type": this._contentType,
+      },
+    };
+
+    if (body) info.body = JSON.stringify(body);
+    if (debug) console.log(`call info:`);
+    if (debug) console.log(info);
+
+    return { url: url, info: info };
+  }
+
+  request(routeData, body) {
+    const requestInfo = this._buildRequest(routeData, body);
+    return fetch(requestInfo.url, requestInfo.info)
+      .then((res) => {
+        if (res.ok) {
+          return res.json();
+        }
+        return Promise.reject(`API request Error: ${res.status}`);
+      })
+      .catch((err) => {
+        console.error(err); // log the error to the console
+      });
+  }
+
+  requestDelete(cardId) {
+    return fetch(this._baseUrl + routes.deleteCardById.route + cardId, {
+      method: "DELETE",
+      headers: {
+        authorization: "0ba3486c-b117-4196-b741-be8eda3e197d",
+        "Content-Type": "application/json",
```

Comment:
According to the task you need to pass full `headers` into the constructor of `Api`. See here https://snipboard.io/pRdaAZ.jpg . And in the code you need to use only: `headers: this._headers` (not to duplicate the code)